### DISCUSSION TITLE
SIP Integration mechanism

### DISCUSSION THREAD

**Discussion by giangndm** [8xFF/atm0s-media-server#365] [Ideas]
SIP system is very complex with a lot of config and customization then I think we should have some general way to integrate with SIP system. Which need:

- Take advantage of atm0s-media
- Don't make amt0s-media too complex
- Allow user can implement our own logic

@marverlous811 give some idea about how to implement a RTPEngine compatible server for integrating with other SIP signaling library like: https://drachtio.org/ . With this method atm0s media only take care about RTP flow

Currently he working on PR: https://github.com/8xFF/atm0s-media-server/pull/359

**Comment by giangndm** [comment]
I just merged first version of SIP integration. More info can found here: [SIP Integrate](https://github.com/8xFF/atm0s-media-server/blob/master/docs/getting-started/quick-start/sip.md)

Thank @marverlous811  for great contribution.

Next we will improve sample: https://github.com/8xFF/atm0s-media-sip-call-sample

**Reply by dhilipsiva** [reply]
Just sharing this assuming you might find this interesting: https://github.com/Televiska/rsip

**Reply by giangndm** [reply]
@dhilipsiva I used RSip in previous version, but it only SIP library then we need to build up PBX system from scratch. At this time I don't have enough time but it look very promised.
